### PR TITLE
E2E: retries in CopyConfigMap func

### PR DESCRIPTION
**Pull request by jackfrancis**
**What type of PR is this?**

/kind flake

**What this PR does / why we need it**:

In response to an observed E2E flake:

https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-cluster-api-provider-azure-e2e-main/1617762815735500800

Let's retry the get/create flow that actually implements "copy" for us, as both calls are subject to transient errors during transmission/delivery.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/helpers.go
Comment (CecileRobertMichon) [inline test/e2e/helpers.go:1024]: I see we have two patterns in the code:

1. The most common:

```
Eventually(func() error {
  err := doThing()
  return err
}, intervals).Should(Succeed())
```

2. What seems to have been introduced more recently in e2e retries, and this:

```
Eventually(func(g Gomega) {
  g.Expect(doThing()).To(Succeed())
}, intervals).Should(Succeed())
```

Are there any reasons for preferring one over the other?

Comment (jackfrancis) [reply test/e2e/helpers.go:1024]: Thanks for bringing this up, I think that more important than preferring one over the other (IMO neither option is materially better or worse) is consistency.

That said, I prefer the 2nd as passing in the Gomega object allows us to continue to express our test flows in idiomatic ginkgo (though arguably not idiomatic go but I digress) and not go back and forth between ginkgo and vanilla go in a single func.

Would love to hear if any other maintainers have ideas, it would be trivial (and IMO beneficial) to do a scrub of our E2E code and standardize to one pattern or the other.

cc @mboersma @devigned

Comment (nojnhuh) [reply test/e2e/helpers.go:991]: I'm also +1 for preferring the `func (g Gomega)` flavor since I think that makes things a bit more internally consistent. I think the bar for breaking that rule should be pretty low though.

Comment (CecileRobertMichon) [reply test/e2e/helpers.go:991]: I don't think this conversation is blocking this PR, the inconsistency already exists before this change, so feel free to move it to an issue if that's easier

Comment (jackfrancis) [reply test/e2e/helpers.go:991]: Since it's easy work (Yes, Please!) I'll do a PR that sets this as the standard pattern in `test/e2e` after this merges.

**Review by mboersma — APPROVED** [review]
/lgtm
/approve

**Comment by jackfrancis** [comment]
My vote is to not worry about manual cherry-picks as we don't see this underlying flake often enough to warrant that add'l effort.

**Pull request merged**
